stack trace with ibm_pi_dhcps data resource

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

```
~> terraform version
Terraform v1.2.5
on linux_amd64
```
### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_pi_dhcps (data)

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# query dhcp services, match a name

locals {
  ids = data.ibm_pi_dhcps.dhcps.servers[*].network_id
  names = data.ibm_pi_dhcps.dhcps.servers[*].network_name
  dhcp_id_from_name = matchkeys(local.ids, local.names, ["dhcpserver"])[0]
}

provider "ibm" {
  ibmcloud_api_key = "..."
  region           = "tor"
  zone             = "tor01"
}

data "ibm_pi_dhcps" "dhcps" {
  pi_cloud_instance_id = "..."
}

output dhcps {
  value = data.ibm_pi_dhcps.dhcps
}

output dhcp_id {
  value = local.dhcp_id_from_name
}

```
Note: this may be incorrect TF, but, it does panic so I figure even if I have errors, it shouldn't panic, so I'm uploading it as-is.

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

```
│   with data.ibm_pi_dhcps.dhcps,
│   on main.tf line 16, in data "ibm_pi_dhcps" "dhcps":
│   16: data "ibm_pi_dhcps" "dhcps" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-ibm_v1.44.1 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2426f8f]

goroutine 131 [running]:
github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/power.dataSourceIBMPIDhcpServersRead(0x37a2788, 0xc000488d80, 0xc000151280, 0x31f4680, 0xc0004b9b00, 0xc00133ee20, 0x616d65686373, 0x40e198)
        github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/power/data_source_ibm_pi_dhcps.go:101 +0x36f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0009336c0, 0x37a27c0, 0xc0013487e0, 0xc000151280, 0x31f4680, 0xc0004b9b00, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:724 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc0009336c0, 0x37a27c0, 0xc0013487e0, 0xc000151180, 0x31f4680, 0xc0004b9b00, 0xc0004b9b00, 0xc000151180, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:943 +0x15c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc0005655d8, 0x37a2718, 0xc0013487e0, 0xc001342340, 0x32217e1, 0x12, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/grpc_provider.go:1179 +0x5b6
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc00009c640, 0x37a27c0, 0xc001348000, 0xc00134c000, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:657 +0x410
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler(0x3118b60, 0xc00009c640, 0x37a27c0, 0xc001348000, 0xc000488840, 0x0, 0x37a27c0, 0xc001348000, 0xc0012100c0, 0x60)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00019e700, 0x37be458, 0xc000dde4e0, 0xc001627c20, 0xc000db3ad0, 0x48bc850, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1282 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00019e700, 0x37be458, 0xc000dde4e0, 0xc001627c20, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1619 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00003f410, 0xc00019e700, 0x37be458, 0xc000dde4e0, 0xc001627c20)
        google.golang.org/grpc@v1.45.0/server.go:921 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.45.0/server.go:919 +0x1fd

Error: The terraform-provider-ibm_v1.44.1 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```

### Expected Behavior

<!--- What should have happened? --->
no panic

### Actual Behavior

panic

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/ibm/service/power/data_source_ibm_pi_dhcp.go
+++ b/ibm/service/power/data_source_ibm_pi_dhcp.go
@@ -109,10 +109,15 @@ func dataSourceIBMPIDhcpRead(ctx context.Context, d *schema.ResourceData, meta i
 	// set attributes
 	d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, *dhcpServer.ID))
 	d.Set(Attr_DhcpID, *dhcpServer.ID)
-	d.Set(Attr_DhcpNetworkDeprecated, *dhcpServer.Network.ID)
-	d.Set(Attr_DhcpNetworkID, *dhcpServer.Network.ID)
-	d.Set(Attr_DhcpNetworkName, *dhcpServer.Network.Name)
 	d.Set(Attr_DhcpStatus, *dhcpServer.Status)
+
+	if dhcpServer.Network != nil {
+		dhcpNetwork := dhcpServer.Network
+		d.Set(Attr_DhcpNetworkDeprecated, *dhcpNetwork.ID)
+		d.Set(Attr_DhcpNetworkID, *dhcpNetwork.ID)
+		d.Set(Attr_DhcpNetworkName, *dhcpNetwork.Name)
+	}
+
 	if dhcpServer.Leases != nil {
 		leaseList := make([]map[string]string, len(dhcpServer.Leases))
 		for i, lease := range dhcpServer.Leases {

--- a/ibm/service/power/data_source_ibm_pi_dhcps.go
+++ b/ibm/service/power/data_source_ibm_pi_dhcps.go
@@ -97,11 +97,14 @@ func dataSourceIBMPIDhcpServersRead(ctx context.Context, d *schema.ResourceData,
 	servers := make([]map[string]interface{}, 0, len(dhcpServers))
 	for _, dhcpServer := range dhcpServers {
 		server := map[string]interface{}{
-			Attr_DhcpID:                *dhcpServer.ID,
-			Attr_DhcpNetworkDeprecated: *dhcpServer.Network.ID,
-			Attr_DhcpNetworkID:         *dhcpServer.Network.ID,
-			Attr_DhcpNetworkName:       *dhcpServer.Network.Name,
-			Attr_DhcpStatus:            *dhcpServer.Status,
+			Attr_DhcpID:     *dhcpServer.ID,
+			Attr_DhcpStatus: *dhcpServer.Status,
+		}
+		if dhcpServer.Network != nil {
+			dhcpNetwork := dhcpServer.Network
+			server[Attr_DhcpNetworkDeprecated] = *dhcpNetwork.ID
+			server[Attr_DhcpNetworkID] = *dhcpNetwork.ID
+			server[Attr_DhcpNetworkName] = *dhcpNetwork.Name
 		}
 		servers = append(servers, server)
 	}

--- a/ibm/service/power/resource_ibm_pi_dhcp.go
+++ b/ibm/service/power/resource_ibm_pi_dhcp.go
@@ -196,10 +196,15 @@ func resourceIBMPIDhcpRead(ctx context.Context, d *schema.ResourceData, meta int
 	// set attributes
 	d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, *dhcpServer.ID))
 	d.Set(Attr_DhcpID, *dhcpServer.ID)
-	d.Set(Attr_DhcpNetworkDeprecated, *dhcpServer.Network.ID)
-	d.Set(Attr_DhcpNetworkID, *dhcpServer.Network.ID)
-	d.Set(Attr_DhcpNetworkName, *dhcpServer.Network.Name)
 	d.Set(Attr_DhcpStatus, *dhcpServer.Status)
+
+	if dhcpServer.Network != nil {
+		dhcpNetwork := dhcpServer.Network
+		d.Set(Attr_DhcpNetworkDeprecated, *dhcpNetwork.ID)
+		d.Set(Attr_DhcpNetworkID, *dhcpNetwork.ID)
+		d.Set(Attr_DhcpNetworkName, *dhcpNetwork.Name)
+	}
+
 	if dhcpServer.Leases != nil {
 		leaseList := make([]map[string]string, len(dhcpServer.Leases))
 		for i, lease := range dhcpServer.Leases {

